submissions/chris-voitova/friends-app/script.js

```diff
@@ -0,0 +1,199 @@
+const url = "https://randomuser.me/api/?results=16";
+const usersContainer = document.querySelector(".users-container");
+
+const initState = {
+  maleGender: false,
+  femaleGender: false,
+  allGenders: true,
+};
+
+let state = Object.assign({}, initState);
+
+const errorMessage = (message) => `<p class="error-message">${message}</p>`;
+
+const userTemplate = (photo, name, email, gender, age) => {
+  return `
+   <div class="user">
+      <img
+         class="user__photo"
+         src=${photo}
+         alt="user-photo"
+      />
+      <div class="user__info">
+         <h3 class="user__name">${name}</h3>
+         <a class="user__email" href="mailto:${email}">
+           ${email}
+         </a>
+         <span class="user__gender">gender: ${gender}</span>
+         <span class="user__age">age: ${age}</span>
+      </div>
+   </div>`;
+};
+
+async function getUsers() {
+  try {
+    const response = await fetch(url);
+    const json = await response.json();
+    return json.results;
+  } catch {
+    usersContainer.innerHTML += errorMessage("oops :( users fetch failed");
+  }
+}
+
+async function app() {
+  const initUsers = await getUsers();
+  const users = initUsers.slice();
+  renderUsers(users);
+  handleSearch(users);
+  handleFilterByGender(users, initUsers);
+  handleSort(users, "#age-sort", sortByAge, [
```

Comment:
Such arguments is a bad idea in JS.

First of all, it is not clear what `#age-sort` means. As well as last array.

If you want pass such stuff, wrap it in the object with properties - in such a way you will name your arguments.